pkg/controllers/status/work_status_controller_test.go

```diff
@@ -757,156 +703,151 @@ func newWork(workName, workNs string, raw []byte, deletionTimestampIsZero ...boo
-			} else {
-				c.ClusterDynamicClientSetFunc = util.NewClusterDynamicClientSet
-				c.Client = fake.NewClientBuilder().WithScheme(gclient.NewSchema()).WithObjects(
-					&clusterv1alpha1.Cluster{
-						ObjectMeta: metav1.ObjectMeta{Name: "cluster"},
-						Spec: clusterv1alpha1.ClusterSpec{
-							APIEndpoint:                 "https://127.0.0.1",
-							SecretRef:                   &clusterv1alpha1.LocalSecretReference{Namespace: "ns1", Name: "secret1"},
-							InsecureSkipTLSVerification: true,
-						},
-						Status: clusterv1alpha1.ClusterStatus{
-							Conditions: []metav1.Condition{
-								{
-									Type:   clusterv1alpha1.ClusterConditionReady,
-									Status: metav1.ConditionTrue,
-								},
-							},
-						},
-					},
-					&corev1.Secret{
-						ObjectMeta: metav1.ObjectMeta{Namespace: "ns1", Name: "secret1"},
-						Data:       map[string][]byte{clusterv1alpha1.SecretTokenKey: []byte("token")},
-					}).Build()
-			}
-
-			informerManager, err := c.getSingleClusterManager(cluster)
-
-			if tt.expectInformer {
-				assert.NotEmpty(t, informerManager)
-			} else {
-				assert.Empty(t, informerManager)
-			}
-
-			if tt.expectError {
-				assert.NotEmpty(t, err)
-			} else {
-				assert.Empty(t, err)
-			}
-		})
-	}
-}
-
-func TestWorkStatusController_recreateResourceIfNeeded(t *testing.T) {
-	c := WorkStatusController{
-		Client:                      fake.NewClientBuilder().WithScheme(gclient.NewSchema()).WithObjects(newCluster("cluster", clusterv1alpha1.ClusterConditionReady, metav1.ConditionTrue)).Build(),
-		InformerManager:             genericmanager.GetInstance(),
-		PredicateFunc:               helper.NewClusterPredicateOnAgent("test"),
-		ClusterDynamicClientSetFunc: util.NewClusterDynamicClientSetForAgent,
-		ClusterCacheSyncTimeout:     metav1.Duration{},
-		RateLimiterOptions:          ratelimiterflag.Options{},
-	}
-
-	raw := []byte(`{"apiVersion":"v1","kind":"Pod","metadata":{"name":"pod","namespace":"default"}}`)
-	work := newWork("work", "default", raw)
-
-	obj := &unstructured.Unstructured{
-		Object: map[string]interface{}{
-			"apiVersion": "v1",
-			"kind":       "Pod",
-			"metadata": map[string]interface{}{
-				"name":      "pod1",
-				"namespace": "default",
-				"labels": map[string]interface{}{
-					workv1alpha1.WorkNamespaceLabel: "karmada-es-cluster",
-				},
-			},
-		},
-	}
-
-	key, _ := generateKey(obj)
-
-	fedKey, ok := key.(keys.FederatedKey)
-	if !ok {
-		t.Fatalf("Invalid key, key: %v", key)
-	}
-
-	t.Run("normal case", func(t *testing.T) {
-		err := c.recreateResourceIfNeeded(work, fedKey)
-		assert.Empty(t, err)
-	})
-
-	t.Run("failed to UnmarshalJSON", func(t *testing.T) {
-		work.Spec.Workload.Manifests[0].RawExtension.Raw = []byte(`{"apiVersion":"v1","kind":"Pod","metadata":{"name":"pod","namespace":"default"}},`)
-		err := c.recreateResourceIfNeeded(work, fedKey)
-		assert.NotEmpty(t, err)
-	})
-}
+// TODO: Move to resourceinformer_test
```

Comment:
Can the move be completed in the current PR?

Reply:
Done. And I also added some unit test for `execution controller`.